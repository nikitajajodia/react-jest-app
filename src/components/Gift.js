import React from 'react';
import PropTypes from 'prop-types';

class Gift extends React.Component {
  constructor() {
    super();
    this.state = {
      person: '',
      present: '',
    };
  }

  render() {
    const {
      person, present,
    } = this.state;
    const {
      removeGift,
      gift,
    } = this.props;
    console.log(person, present);
    return (
      <div>
        <form>
          Person
          <input
            type="text"
            className="input-person"
            onChange={event => this.setState({ person: event.target.value })}
          />
          Present
          <input
            type="text"
            className="input-present"
            onChange={event => this.setState({ present: event.target.value })}
          />
        </form>
        <button
          onClick={() => removeGift(gift.id)}
          type="button"
          className="remove-gift"
        >
            Remove Gift
        </button>
      </div>
    );
  }
}

export default Gift;

Gift.propTypes = {
  removeGift: PropTypes.func.isRequired,
  gift: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};
