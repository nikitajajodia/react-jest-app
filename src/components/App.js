import React from 'react';
import Gift from './Gift';

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      gifts: [],
    };
  }

  addGift = () => {
    const {
      gifts,
    } = this.state;

    const ids = gifts.map(gift => gift.id);
    const maxId = ids.length > 0 ? Math.max(...ids) : 0;

    gifts.push({
      id: maxId + 1,
    });

    this.setState({
      gifts,
    });
  }

  render() {
    const {
      gifts,
    } = this.state;
    return (
      <div>
        <h2>
          Gift Giver
        </h2>
        <div className="gift-list">
          {
            gifts.map(gift => (
              <div key={gift.id}>
                <Gift />
              </div>
            ))
          }
        </div>
        <button className="btn-add" onClick={this.addGift} type="button">
          Add Gift
        </button>
      </div>
    );
  }
}

export default App;
