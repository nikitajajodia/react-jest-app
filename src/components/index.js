import Header from './Header';
import Footer from './Footer';
import App from './App';
import Gift from './Gift';

export {
  Header,
  Footer,
  App,
  Gift,
};
