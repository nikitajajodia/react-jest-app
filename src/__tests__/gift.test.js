import React from 'react';

import { Gift } from '../components';

describe('Gift-->', () => {
  const gift = shallow(<Gift />);

  it('renders properly', () => {
    expect(gift).toMatchSnapshot();
  });

  it('initialises a person and present in that state', () => {
    expect(gift.state()).toEqual({ person: '', present: '' });
  });

  describe('when typing into the person input', () => {
    const person = 'Unlce';
    beforeEach(() => {
      gift.find('.input-person').simulate('change', { target: { value: person } });
    });

    it('updates the person in state', () => {
      expect(gift.state().person).toEqual(person);
    });
  });

  describe('when typing into the present input', () => {
    const present = 'Golf Clubs';
    beforeEach(() => {
      gift.find('.input-present').simulate('change', { target: { value: present } });
    });

    it('updates the present in state', () => {
      expect(gift.state().present).toEqual(present);
    });
  });
});
