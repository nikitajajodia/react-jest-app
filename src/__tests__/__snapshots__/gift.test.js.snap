// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift--> renders properly 1`] = `
<div>
  <form>
    Person
    <input
      className="input-person"
      onChange={[Function]}
      type="text"
    />
    Present
    <input
      className="input-present"
      onChange={[Function]}
      type="text"
    />
  </form>
  <button
    className="remove-gift"
    onClick={[Function]}
    type="button"
  >
    Remove Gift
  </button>
</div>
`;
